### PR TITLE
Support for OpenSSH new key file format 

### PR DESCRIPTION
**Pull request by hierynomus**
The new format is used for ed-25519 keys.

Only support for non-encrypted keys currently.

### CODE REVIEW

**Comment by hierynomus** [comment]
@dkocher Can you test it out?

**Comment by dkocher** [comment]
@hierynomus Will do today!

**Comment by hierynomus** [comment]
It's a bit later then I wanted, but it took some experimenting ;)

File: src/main/java/com/hierynomus/sshj/userauth/keyprovider/OpenSSHKeyV1KeyFile.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            throw new IOException("We don't support having more than 1 key in the file (yet).");
+        }
+        PublicKey publicKey = readPublicKey(new PlainBuffer(keyBuffer.readBytes())); // string publickey1
+        PlainBuffer privateKeyBuffer = new PlainBuffer(keyBuffer.readBytes()); // string (possibly) encrypted, padded list of private keys
+        if ("none".equals(cipherName)) {
+            logger.debug("Reading unencrypted keypair");
+            return readUnencrypted(privateKeyBuffer, publicKey);
+        } else {
+            logger.info("Keypair is encrypted with: " + cipherName + ", " + kdfName + ", " + kdfOptions);
+            throw new IOException("Cannot read encrypted keypair with " + cipherName + " yet..");
```
Comment (dkocher) [inline src/main/java/com/hierynomus/sshj/userauth/keyprovider/OpenSSHKeyV1KeyFile.java]: What additional work is required to support encrypted keys?

**Comment by dkocher** [comment]
I have tested this with a private key of mine and it works as advertised.

File: src/main/java/com/hierynomus/sshj/userauth/keyprovider/OpenSSHKeyV1KeyFile.java (hunk 1 above)
Comment (hierynomus) [inline src/main/java/com/hierynomus/sshj/userauth/keyprovider/OpenSSHKeyV1KeyFile.java]: I have to figure out how the 'altered' bcrypt implementation works... i.e. seems non-trivial at the moment.

**Comment by hierynomus** [comment]
Yay! If you can think of a better class name I'll rename it. Else I'll leave that for a moment.

Going through this code again, it could use a lot of cleanup. I want to split off the keyfile format from the keytype stored in there. Will do that separately though.

**Pull request merged**